apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    type SuiSignMessageOutput,
+    type SuiSignMessageOptions,
+    type SuiSignAndExecuteTransactionOptions,
+} from '@mysten/wallet-standard';
+
+import type {
+    MoveCallTransaction,
+    SignableTransaction,
+    SignedTransaction,
+    SuiAddress,
+    SuiMoveNormalizedFunction,
+    SuiTransactionResponse,
+    UnserializedSignableTransaction,
+} from '@mysten/sui.js';
+
+export type TransactionDataType =
+    | {
+          type: 'v2';
+          justSign?: boolean;
+          //   TODO: Support transaciton builder string
+          //   data: SignableTransaction | string;
+          data: SignableTransaction;
+          options?: SuiSignAndExecuteTransactionOptions;
+          account: SuiAddress;
+      }
+    | { type: 'move-call'; data: MoveCallTransaction; account: SuiAddress }
```

Comment:
While you're here, if you rebase, you can remove the `move-call` and `serialized-move-call`.

Reply:
I forgot to do this earlier 😅

Reply:
This requires many changes in the end will do it in another PR to be reviewed properly

Reply:
https://mysten.atlassian.net/browse/APPS-611